pkg/model/bootstrapscript.go

```diff
@@ -71,6 +72,40 @@ func (b *BootstrapScript) ResourceNodeUp(ig *kops.InstanceGroup) (*fi.ResourceHo
 			}
 			return ""
 		},
+
+		"ProxyEnv": func() string {
+			scriptSnippet := ""
+
+			if ps != nil && ps.HTTPProxy.Host != "" {
+				httpProxyUrl := "http://"
+				if ps.HTTPProxy.User != "" {
+
+					httpProxyUrl += ps.HTTPProxy.User
+					if ps.HTTPProxy.Password != "" {
+						httpProxyUrl += "@" + ps.HTTPProxy.Password
+					}
+				}
+				httpProxyUrl += ps.HTTPProxy.Host + ":" + strconv.Itoa(ps.HTTPProxy.Port)
+				scriptSnippet =
+					"export http_proxy=" + httpProxyUrl + "\n" +
+						"export https_proxy=${http_proxy}\n" +
```

Comment:
It's only needed if the user has seperate servers or ports for http and https proxies.  If you only specify one, most clients will try to used that one for both.  I'm guessing the more common configuration is to have one proxy and port for both protocols. However, to support the full gamut of possible configurations, you would need both, and also ftp_proxy as well.  My first iteration had this, but I removed it since it wasn't needed, and thought it could be added back if someone requested it.